### PR TITLE
Fix attribute routing error message related to replace tokens 

### PR DESCRIPTION
**Pull request by kichalla**
Related to https://github.com/aspnet/Routing/issues/373

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.Core/Resources.resx [hunk 1]
```diff
@@ -199,7 +199,7 @@
     <comment>{1} is the specific error message.</comment>
   </data>
   <data name="AttributeRoute_TokenReplacement_ReplacementValueNotFound" xml:space="preserve">
-    <value>While processing template '{0}', a replacement value for the token '{1}' could not be found. Available tokens: '{2}'.</value>
+    <value>While processing template '{0}', a replacement value for the token '{1}' could not be found. Available tokens: '{2}'. To use a '[' or a ']' as a literal string in a route or within a constraint use '[[' or ']]' instead.</value>
```
Comment (dougbu) [inline src/Microsoft.AspNetCore.Mvc.Core/Resources.resx]: The repeated "a" is a bit awkward and the sentence can be simplified. Suggest `To use a literal '[' or ']' in ...`. Also, add a comma after "constraint".

File: test/Microsoft.AspNetCore.Mvc.Core.Test/ApplicationModel/AttributeRouteModelTests.cs [hunk 2]
```diff
@@ -175,6 +175,32 @@ public void ReplaceTokens_InvalidFormat(string template, Dictionary<string, stri
         }
 
         [Fact]
+        public void ReplaceTokens_UnknownValue_InConstraints()
```
Comment (dougbu) [inline test/Microsoft.AspNetCore.Mvc.Core.Test/ApplicationModel/AttributeRouteModelTests.cs]: Instead make the existing `ReplaceTokens_UnknownValue()` a `[Theory]`. You've already made one of the main required changes there.

File: test/Microsoft.AspNetCore.Mvc.Core.Test/ApplicationModel/AttributeRouteModelTests.cs [hunk 3]
```diff
@@ -570,6 +597,8 @@ public static IEnumerable<object[]> ReplaceTokens_InvalidFormatValuesData
                     "An unescaped '[' token is not allowed inside of a replacement token. Use '[[' to escape."
                 };
 
+                // ("api/departments/{deptName:regex(^[a-zA-Z]{1}[a-zA-Z0-9_]*$)}"
```
Comment (dougbu) [inline test/Microsoft.AspNetCore.Mvc.Core.Test/ApplicationModel/AttributeRouteModelTests.cs]: Huh? Think this should be reverted.

File: test/Microsoft.AspNetCore.Mvc.Core.Test/ApplicationModel/AttributeRouteModelTests.cs [hunk 4]
```diff
@@ -187,9 +213,10 @@ public void ReplaceTokens_UnknownValue()
             };
 
             var expected =
-                "While processing template '[area]/[controller]/[action2]', " +
+                $"While processing template '{template}', " +
                 "a replacement value for the token 'action2' could not be found. " +
```
Comment (dougbu) [inline test/Microsoft.AspNetCore.Mvc.Core.Test/ApplicationModel/AttributeRouteModelTests.cs]: Pass `"a-zA-Z"` versus `"action2"` into the new theory.

**Comment by kichalla** [comment]
🆙 📅

File: test/Microsoft.AspNetCore.Mvc.Core.Test/ApplicationModel/AttributeRouteModelTests.cs [hunk 5]
```diff
@@ -174,11 +174,12 @@ public void ReplaceTokens_InvalidFormat(string template, Dictionary<string, stri
             Assert.Equal(expected, ex.Message);
         }
 
-        [Fact]
-        public void ReplaceTokens_UnknownValue()
+        [Theory]
+        [InlineData("[area]/[controller]/[action]/{deptName:regex(^[a-zA-Z]{1}[a-zA-Z0-9_]*$)}", "a-zA-Z")]
+        [InlineData("[area]/[controller]/[action2]", "action2")]
```
Comment (dougbu) [inline test/Microsoft.AspNetCore.Mvc.Core.Test/ApplicationModel/AttributeRouteModelTests.cs]: 😀

**Pull request merged**